types/layui-src/Layui.d.ts

```diff
@@ -1,381 +1,381 @@
-    /**
-     *  用于将一段 URL 链接中的 pathname、search、hash 属性值进行对象化处理	<br/>&nbsp;
-     *  参数： href 可选。若不传，则自动读取当前页面的 url（即：location.href）	<br/>&nbsp;
-     *  示例：let url = layui.url();
-     * @param [href]  http(s)地址
-     */
-    url(href?: string): {
-        hash: Layui.UrlHash;
-        pathname: string[];
-        search: object;
-    };
-    // use function
-    // https://www.layui.com/doc/
-    // https://www.layui.com/doc/base/infrastructure.html
-    // https://www.layui.com/doc/base/modules.html
-    /**
-     * 使用特定模块
-     * @param [mods]  内置或自定模块名 (若模块不存在则抛js错误，callback不会执行)
-     * @param [callback]  回调函数
-     * @param [exports]  数组，存储对 mods解析后加载的模块
-     */
-    use<K extends keyof Layui.LayuiModuleMap>(
-        mods: K,
-        callback: (this: Layui, module: Layui.LayuiModuleMap[K]) => any,
-        exports?: any[],
-    ): { v: string };
-    /**
-     * 使用自定模块
-     * @param [mods]  自定义模块，非内置模块
-     * @param [callback]  回调函数
-     * @param [exports]  数组，存储对 mods解析后加载的模块
-     */
-    use(mods: string, callback: (this: Layui, module: any) => any, exports?: any[]): { v: string };
-
-    /**
-     * 使用多模块
-     * @param [mods]  内置或自定模块名 (若模块不存在则抛js错误，callback不会执行)
-     * @param [callback]  回调函数 	<br/>&nbsp;
-     *  1、不建议callback中设置参数因为没有TS约束，可用layui.xx调用具体模块;	<br/>&nbsp;
-     *  2、手动在callback回调中指定类型比如 util:layui.Util
-     * @param [exports] 暴露出挂载的对象
-     */
-    use(mods: string[], callback: (this: Layui, ...module: any) => any, exports?: any[]): { v: string };
-
-    /**
-     * 使用特定模块
-     * @param [callback]  回调函数， 从 layui 2.6 开始，首个参数是callback函数，则表示引用所有内置模块到layui.xx<br/>&nbsp;
-     * @param [exports]  无任何用途，可不传
-     */
-    use(callback: (this: Layui, module: { config: object; time: number }) => any, exports?: any[]): { v: string };
-
-    /**
-     * 代码高亮
-     * https://www.layui.com/doc/modules/code.html
-     * @param [option] 基础参数
-     */
-    code(option?: Layui.CodeOption): void;
-}
+interface Layui {
+    $: JQueryStatic;
+    carousel: Layui.Carousel;
+    colorpicker: Layui.ColorPicker;
+    dropdown: Layui.DropDown;
+    element: Layui.Element;
+    flow: Layui.Flow;
+    form: Layui.Form;
+    global: {};
+    jquery: JQueryStatic;
+    lay: Layui.Lay;
+    laydate: Layui.Laydate;
+    layedit: Layui.Layedit;
+    layer: Layui.Layer;
+    laypage: Layui.Laypage;
+    laytpl: Layui.Laytpl;
+    "layui.all": string;
+    rate: Layui.Rate;
+    slider: Layui.Slider;
+    table: Layui.Table;
+    transfer: Layui.Transfer;
+    tree: Layui.Tree;
+    upload: Layui.Upload;
+    util: Layui.Util;
+    v: string;
+
+    // https://www.layui.com/doc/base/infrastructure.html
+    /**
+     * 静态属性。获得一些配置及临时的缓存信息
+     */
+    cache: Layui.CacheData;
+
+    /**
+     * 内置模块名和外置名称映射
+     */
+    modules: Layui.Modules;
+
+    // -----------------prototype method-------------------------------------↓↓↓↓
+
+    // https://www.layui.com/doc/base/infrastructure.html
+    /**
+     * 对象是否为泛数组结构，实例：	<br/>&nbsp;
```

Comment:
<!--
  1 failure:  The root .editorc...
  0 warning: 
  
  
  DangerID: danger-id-Danger;
  File: types/layui-src/Layui.d.ts;
  Line: 42;
-->
- :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.